blob_io_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright (C) 2022 Yasuhiro Matsumoto <mattn.jp@gmail.com>.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+
+//go:build cgo
+// +build cgo
+
+package sqlite3
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"io"
+	"testing"
+)
+
+// Verify interface implementations
+var _ io.Reader = &SQLiteBlob{}
+var _ io.Writer = &SQLiteBlob{}
+var _ io.Seeker = &SQLiteBlob{}
+var _ io.Closer = &SQLiteBlob{}
+
+func blobTestData(dbname string, rowid int64, blob []byte) (*sql.DB, *SQLiteConn, error) {
+	db, err := sql.Open("sqlite3", "file:"+dbname+"?mode=memory&cache=shared")
```

Comment:
As shared cache mode is discouraged by the SQLite authors, I would prefer not to use it in tests so that clients don't get the wrong idea. Instead please use `file:/<dbname>?vfs=memdb`. Note the leading slash in the database name - this is critical for a memdb to be shared across connections in the pool.

Reply:
If you use `:memory:` then you can run into issues due to the connection pooling, as each connection will be using a separate in-memory database. For this reason, `vfs=memdb` is generally preferred.